osf/models/metaschema.py

```diff
@@ -194,14 +179,34 @@ class Meta:
         order_with_respect_to = 'schema'
         unique_together = ('schema', 'registration_response_key')
 
+    INPUT_BLOCK_TYPES = frozenset([
+        ('short-text-input', 'short-text-input'),
+        ('long-text-input', 'long-text-input'),
+        ('file-input', 'file-input'),
+        ('contributors-input', 'contributors-input'),
+        ('single-select-input', 'single-select-input'),
+        ('multi-select-input', 'multi-select-input'),
+        ('select-input-option', 'select-input-option'),
+        ('select-other-option', 'select-other-option'),
+    ])
+    UNGROUPED_BLOCK_TYPES = frozenset([
+        ('page-heading', 'page-heading'),
+        ('section-heading', 'section-heading'),
+        ('subsection-heading', 'subsection-heading'),
+        ('paragraph', 'paragraph'),
+        ('question-label', 'question-label'),
+    ])
+
+    BLOCK_TYPES = UNGROUPED_BLOCK_TYPES.union(INPUT_BLOCK_TYPES)
```

Comment:
1) The tuple-ness goes unused for INPUT_BLOCK_TYPES and UNGROUPED_BLOCK_TYPES
2) You've changed the logic below so you don't really need `UNGROUPED_BLOCK_TYPES` -- also, 'question_label' does get a `schemablock_group_key`, so this is incorrectly named, anyway.
3) No particular reason to change the name of SCHEMABLOCK_TYPES

I'd recommend just keeping the `INPUT_BLOCK_TYPES` (but making it *not* a set of tuples), leaving `SCHEMABLOCK_TYPES` alone, and removing `UNGROUPED_BLOCK_TYPES`.

Option two would be renaming `UNGROUPED_BLOCK_TYPES` to something like `DISPLAY_BLOCK_TYPES`, making it and `INPUT_BLOCK_TYPES` both flat sets, and constructing `SCHEMABLOCK_TYPES` as `[(type, type) for type in INPUT_BLOCK_TYPES.union(DISPLAY_BLOCK_TYPES)]` -- maybe with the explicit comment that this is used to generate CharField choices, since it's weird that this is a list of tuples, anyway.